src/ConfigFileParser.cpp

```diff
@@ -0,0 +1,534 @@
+    configServer.serverName = "";
+    configServer.root = "html";
+    configServer.listen.insert(std::make_pair("127.0.0.1", 80));
+    configServer.locationIndex = 0;
+    configServer.maxBodySize = 1;
+    configServer.errorPage = std::map<unsigned short, std::string>();
+    configServer.locationIndex = 0;
+    configServer.locations = std::vector<Location>();
+
+    m_configFile.servers.push_back(configServer);
+}
+
+/**
+ * @brief Initializes the Location object with default values
+ *
+ * @param location The Location object to initialize
+ */
+void ConfigFileParser::initializeLocation(Location& location)
+{
+    const char* validLocationDirectiveNames[] = { "root", "index", "cgi_ext", "cgi_path", "autoindex", "limit_except", "location", "return" };
+    const int validLocationDirectiveNamesSize = sizeof(validLocationDirectiveNames) / sizeof(validLocationDirectiveNames[0]);
+
+    location.path = "";
+    location.root = "html";
+    location.indices = std::vector<std::string>();
+    location.cgiExt = "";
+    location.cgiPath = "";
+    location.isAutoindex = false;
+    location.returns = std::map<unsigned short, std::string>();
+    location.validLocationDirectives = std::vector<std::string>(validLocationDirectiveNames, validLocationDirectiveNames + validLocationDirectiveNamesSize);
+    m_configFile.servers[m_configFile.serverIndex].locations.push_back(location);
+}
+
+/**
+ * @brief Checks if there are no open brackets
+ *
+ * If a opening bracket is found, it is pushed onto the brackets stack
+ * If a closing bracket is found, it is popped from the brackets stack
+ *
+ * At the end the stack should be empty. If not, there are open brackets
+ *
+ * @param configFilePath Path from the config file
+ * @return true If there is minimum one open bracket
+ * @return false If there no open bracket
+ */
+bool ConfigFileParser::isBracketOpen(const std::string& configFilePath) const
+{
+    std::ifstream tmpStream;
+    std::string tmpLine;
+    std::stack<char> brackets;
+
+    tmpStream.open(configFilePath.c_str());
+
+    while (getline(tmpStream, tmpLine)) {
+        for (std::string::const_iterator it = tmpLine.begin(); it != tmpLine.end(); ++it) {
+            if (*it == '{')
+                brackets.push('{');
+            else if (*it == '}' && brackets.empty())
+                return true;
+            else if (*it == '}')
+                brackets.pop();
+        }
+    }
+    return !brackets.empty();
+}
+
+/**
+ * @brief Checks if the line contains a semicolon
+ *
+ * @param line The current line to be checked
+ * @return true If the line does not contain a semicolon
+ * @return false If the line contains a semicolon
+ */
+bool ConfigFileParser::isSemicolonMissing(const std::string& line) const
+{
+    return line.find(';') == std::string::npos && getDirective(line) != "location";
```

Comment:
Exactly. I don't want to check if a semicolon is missing when the directive is "location", because there is no semicolon needed for this case. 

I added additionally clarification in the comments.